### ISSUE TITLE
"x is numpy.nan" always False

### ISSUE THREAD

**Issue by gamask** [serge-sans-paille/pythran#2266]
```
import numpy

# pythran export f(float64)
def f(x):
    return x is numpy.nan
```

If passed numpy.nan this function will return True when not pythranized, and False when pythranized.

**Comment by serge-sans-paille** [comment]
On Thu, Dec 05, 2024 at 11:44:13AM -0800, gamask wrote:
> import numpy
> 
> # pythran export f(float64)
> def f(x):
>     return x is numpy.nan

This is not the correct way to check if some value is a Nan (even in Python!)
Consider the following:

    >>> import numpy as np
    >>> x = np.nan
    >>> y = x
    >>> y = x + 0
    >>> x is y
    False
    >>> np.isnan(x)
    np.True_
    >>> np.isnan(y)
    np.True_
    >>> import numpy as np
    >>> x = np.nan
    >>> x is np.nan
    True
    >>> y = x + 0
    >>> y is np.nan
    False
    >>> np.isnan(x)
    np.True_
    >>> np.isnan(y)
    np.True_

so basically check if a number is a not a number with isnan, check if two
values are bound to the same object with `is`, and check for equality with `==`.

Checking for value equality with `is` is implementation undefined, as showcased
by your Pythran failure :-)

**Comment by gamask** [comment]
Fair enough.
I knew about isnan; I was using "is" in some dumb testing code where I explicitly set some things to nan, and it worked :)
Didn't realize it was undefined behaviour
